healthx/handler_test.go

```diff
@@ -36,6 +36,27 @@ import (
 	"github.com/ory/herodot"
 )
 
+const mockHeaderKey = "middleware-header"
+const mockHeaderValue = "test-header-value"
+
+// MiddlewareHeader is a example logger for the below tests
+type MiddlewareHeader struct {
+	handler http.Handler
+	t       *testing.T
+}
+
+func (l *MiddlewareHeader) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add(mockHeaderKey, mockHeaderValue)
+	l.handler.ServeHTTP(w, r)
+}
+
+func loggerMiddleware(t *testing.T) func(n http.Handler) http.Handler {
+	return func(n http.Handler) http.Handler {
+		logger := &MiddlewareHeader{handler: n, t: t}
+		return logger
+	}
+}
```

Comment:
I don't think you need the `MiddlewareHeader` struct here, we could just wrap it in the middleware immediately functionally and then just do an assertion as well to make sure the request is coming from the right source, e.g. the method is a GET, or pass the expected values. You can also still set the mock header values to assert later that it did indeed execute.

```go
writer.Header().Add(mockHeaderKey, mockHeaderValue)
```

I would just do 

```go
testMiddleware := func(t *testing.T, expectedStuff func(*testing.T, http.ResponseWriter,*http.Request)) func(next http.Handler) http.Handler { 
  //call expectedStuff
  //do middleware stuff
  })
}
```

I would also call this something else instead of `loggerMiddleware`. Maybe `testMiddleware` since it is a test middleware for our health and version routes.

Check out https://github.com/ory/x/blob/4bf4fdd378ce28464cc6374314b9026b2291a486/proxy/proxy.go#L201-L217 for inspiration on how to do this.

Reply:
> I would just do

Um, while it works , in my opinion the one with the type struct is a little more readable?
the function as a param and return do add a lot of noise to the simple definition of a http.handler

Reply:
Here's what the working code would look like with the above change 
```go
```

Reply:
you only need one function :) not 2

Reply:
> Um, while it works , in my opinion the one with the type struct is a little more readable?

I don't agree, with the struct you have way more lines of code and you still need the middleware function anyway. with this change you don't need the struct and can do it in one function that does the assertions (which is what we want to test for in the middleware) and a test asserting the middleware was used.